src/api/firebase.js

```diff
@@ -104,18 +105,49 @@ export async function addItem(
 	}
 }
 
-export async function updateItem(listToken, itemId, totalPurchases) {
+/**
+ * @param {String} listToken the list token that corresponds to a Firebase collection
+ * @param {Object} itemData the object containing all of the current data for this item from our app's state (destructured in implementation)
+ */
+export async function updateItem(
+	listToken,
+	{ id, dateLastPurchased, dateNextPurchased, dateCreated, totalPurchases },
+) {
 	/**
 	 * TODO: Fill this out so that it uses the correct Firestore function
 	 * to update an existing item. You'll need to figure out what arguments
 	 * this function must accept!
 	 */
+
+	// get current date, the last estimated interval, days since last transaction, and increment total purchases
+	const today = new Date();
+	const lastEstimatedInterval =
+		dateLastPurchased === null
+			? getDaysBetweenDates(dateNextPurchased.toDate(), dateCreated.toDate())
+			: getDaysBetweenDates(
+					dateNextPurchased.toDate(),
+					dateLastPurchased.toDate(),
+			  );
+	const daysSinceLastTransaction =
+		dateLastPurchased === null
+			? getDaysBetweenDates(today, dateCreated.toDate())
+			: getDaysBetweenDates(today, dateLastPurchased.toDate());
+	const newTotalPurcahses = totalPurchases + 1;
```

Comment:
This is just a spelling error, but purchases is misspelled in the const. It says purcahses instead. Otherwise everything looks great! Nice job making this work so smoothly even though you had lots of numbers to keep track of. You make it look easy :)

Reply:
Thanks! Just pushed a fix for the spelling.